app/shared-components/PromoModal/PromoModal.jsx

```diff
@@ -0,0 +1,201 @@
+				break;
+			case PREMIUM:
+				sendMessage('ping', 'promo_modals_decline_insights_upgrade');
+				break;
+			default:
+				break;
+		}
+
+		this.props.actions.showNotification({
+			classes: 'warning',
+			reload: false,
+			text: t('promos_turned_off_notification'),
+		});
+	};
+
+	/**
+	 * @private
+	 * Handle clicks on the download buttons
+	 */
+	_handlePromoTryProductClick = (product, utm_campaign) => {
+		this.props.actions.togglePromoModal();
+
+		let url;
+		switch (product) {
+			case PLUS:
+				// Should we send a plus ping here?
+				// sendMessage('ping', 'promo_modals_plus_upgrade_cta');
+				url = `https://checkout.${DOMAIN}.com/plus?utm_source=gbe&utm_campaign=${utm_campaign}`;
+				break;
+			case PREMIUM:
+				url = `https://ghostery.com/thanks-for-downloading-midnight?utm_source=gbe&utm_campaign=${utm_campaign}`;
+				break;
+			case INSIGHTS:
+				sendMessage('ping', 'promo_modals_insights_upgrade_cta');
+				url = `https://checkout.${DOMAIN}.com/insights?utm_source=gbe&utm_campaign=${utm_campaign}`;
+				break;
+			default:
+		}
+
+		sendMessage('openNewTab', {
+			url,
+			become_active: true,
+		});
+	}
+
+	/**
+	 * @private
+	 * Handle clicks on sign in links in promo modals
+	 */
+	_handlePromoSignInClick = () => {
+		this.props.actions.togglePromoModal();
+		history.push({
+			pathname: '/login',
+		});
+	};
+
+	_handlePromoXClick = (type) => {
+		this.props.actions.togglePromoModal();
+
+		if (type === INSIGHTS) {
+			sendMessage('ping', 'promo_modals_decline_insights_upgrade');
+		} else if (type === PLUS) {
+			sendMessage('ping', 'promo_modals_decline_plus_upgrade');
+		}
+	}
+
+	_renderXButton = (type) => {
+		const XButtonClass = ClassNames({ PlusPromoModal__exitButton: type === PLUS });
+
+		return (
+			<ModalExitButton
+				className={XButtonClass}
+				toggleModal={() => this._handlePromoXClick(type)}
+			/>
+		);
+	};
+
+	renderModalContent() {
+		const { type, loggedIn } = this.props;
+		switch (type) {
+			case INSIGHTS:
+				return (
+					<InsightsPromoModalContent
+						handleGoAwayClick={() => this._handlePromoGoAwayClick(INSIGHTS)}
+						handleTryInsightsClick={() => this._handlePromoTryProductClick(INSIGHTS, 'in_app_upgrade')}
+						handleSignInClick={this._handlePromoSignInClick}
+						{...this.props}
+					/>
+				);
+			case PLUS:
+				return (
+					<PlusPromoModalContent
+						handleGoAwayClick={() => this._handlePromoGoAwayClick(PLUS)}
+						handleTryPlusClick={() => this._handlePromoTryProductClick(PLUS, 'in_app_spring2020')}
+						handleSignInClick={this._handlePromoSignInClick}
+						loggedIn={loggedIn}
+					/>
+				);
+			case PREMIUM:
+				return (
+					<PremiumPromoModalContent
+						handleGoAwayClick={() => this._handlePromoGoAwayClick(PREMIUM)}
+						handleTryMidnightClick={() => this._handlePromoTryProductClick(PREMIUM, 'in_app')}
+						handleGetPlusClick={() => this._handlePromoTryProductClick(PLUS, 'in_app')}
```

Comment:
Should first arg be `PREMIUM`?

Reply:
I think the handleGetPlusClick on the premium promo modal should lead to the checkout page for plus if I'm not mistaken? I wound up reusing the `_handlePromoTryProductClick()` function because it doesn't send UTM params

Reply:
My bad, I forgot that the Premium modal has an option to get Plus instead. Looks good.